### DISCUSSION TITLE
Extend 3D Panel with custom SceneExtensions

### DISCUSSION THREAD

**Discussion by garb4ge** [foxglove/community#795] [Q&A] [answered]
I am trying to add a custom `SceneExtension` to the 3D panel for displaying a map that is not read from messages but from another source such as a REST API that returns points / polygons in a certain frame. Hence, I cannot use a message transformer module since I do not get the map as a message.

As far as I can see there is no chance to extend the 3D panel with custom `SceneExtension`s via classical extensions? Do you see another solution?

Follow-up question: Is it possible to add settings to message converters that can be adjusted via the GUI? E.g. select a subset of features/properties of objects to render from a message that represents a list of objects containing many properties such as velocity, acceleration, type, confidence, ...

**Comment by defunctzombie** [comment, marked as answer]
> Do you see another solution?

Nope - this is not currently possible.

> Is it possible to add settings to message converters that can be adjusted via the GUI?

This is not currently possible.
